### PR TITLE
Validate metric & at backend connection/queries

### PR DESCRIPTION
**Pull request by ntk148v**
Close: #60 
Close: #59

### CODE REVIEW

File: pkg/autoscaler/manager.go [hunk 1]
```diff
@@ -17,7 +17,7 @@ package autoscaler
 import (
 	"context"
 	"encoding/json"
-	"fmt"
+	"github.com/vCloud-DFTBA/faythe/pkg/metrics"
```
Comment (vtdat) [inline pkg/autoscaler/manager.go]: import placing

File: api/cloud.go [hunk 2]
```diff
@@ -69,8 +70,19 @@ func (a *API) registerCloud(w http.ResponseWriter, req *http.Request) {
 			return
 		}
 
+		// Register Backend to registry
+		err := metrics.Register(ops.Monitor.Backend, string(ops.Monitor.Address),
+			ops.Monitor.Username, ops.Monitor.Password)
+		if err != nil {
+			a.respondError(w, apiError{
+				code: http.StatusBadRequest,
+				err:  fmt.Errorf("invalid backend connection: %s", err.Error()),
```
Comment (vtdat) [inline api/cloud.go]: `cannot connect to backend` sounds better?

File: api/healer.go [hunk 3]
```diff
@@ -71,13 +73,23 @@ func (a *API) createHealer(rw http.ResponseWriter, req *http.Request) {
 		return
 	}
 
+	// Validate query format.
+	backend, _ := metrics.GetBackend(a.etcdcli, vars["provider_id"])
+	_, err := backend.QueryInstant(req.Context(), h.Query, time.Now())
+	if err != nil && strings.Contains(err.Error(), "bad_data") {
+		err = fmt.Errorf("invalid query: %s", err.Error())
+		a.respondError(rw, apiError{
+			code: http.StatusInternalServerError,
```
Comment (vtdat) [inline api/healer.go]: should be `StatusBadRequest`?

File: api/scaler.go [hunk 4]
```diff
@@ -75,8 +78,21 @@ func (a *API) createScaler(w http.ResponseWriter, req *http.Request) {
...
 		return
 	}
+
+	// Validate query format.
+	backend, _ := metrics.GetBackend(a.etcdcli, vars["provider_id"])
+	_, err := backend.QueryInstant(req.Context(), s.Query, time.Now())
+	if err != nil && strings.Contains(err.Error(), "bad_data") {
+		err = fmt.Errorf("invalid query: %s", err.Error())
+		a.respondError(w, apiError{
+			code: http.StatusInternalServerError,
```
Comment (vtdat) [inline api/scaler.go]: should be `StatusBadRequest`.

File: api/scaler.go [hunk 5]
```diff
@@ -75,8 +78,21 @@ func (a *API) createScaler(w http.ResponseWriter, req *http.Request) {
...
+			err:  err,
+		})
+		return
```
Comment (vtdat) [inline api/scaler.go]: As discussed, could this handle wrong query that always return no result? 

or it just checks whether query is syntactically correct?

File: api/cloud.go [hunk 6]
```diff
@@ -17,6 +17,7 @@ package api
 import (
 	"encoding/json"
 	"fmt"
+	"github.com/vCloud-DFTBA/faythe/pkg/metrics"
```
Comment (vtdat) [inline api/cloud.go]: import placing

File: api/scaler.go (hunk 5 above)
Comment (ntk148v) [reply api/scaler.go]: Leave a comment in issue: https://github.com/vCloud-DFTBA/faythe/issues/59#issuecomment-566928682

File: api/cloud.go (hunk 6 above)
Comment (vtdat) [reply api/cloud.go]: not yet bro

**Pull request merged**